doc/lin/index.mld

```diff
@@ -0,0 +1,306 @@
+- an encoding of its type which is used to generate arguments to the function.
+
+[val_] is a specialization of [val_freq], giving the same weight to all the
+elements.
+
+Note that the domain specific language brings some static guarantees about the
+type encoding. If we make a mistake, we will know at compile time.
+
+{2 Running the tests}
+
+Now we are set to run our first [qcheck-lin] tests!
+
+We will be testing our library for parallel usage. The functor
+{!module-Lin_domain.Make} takes the [Spec] as arguments and exposes two
+functions:
+
+- {{!Lin_domain.Make.lin_test}lin_test} to build a positive [QCheck] test,
+- {{!Lin_domain.Make.neg_lin_test}lin_test} to build a negative [QCheck] test.
+
+Here, we expect the test to succeed, so we will use the first one.
+
+{[
+module LibDomain = Lin_domain.Make (Spec)
+
+let _ =
+  QCheck_runner.run_tests ~verbose:true
+    [ LibDomain.lin_test ~count:1000 ~name:"Lin parallel tests" ]
+]}
+
+And the test fails...
+
+{[
+$ dune exec ./mutable_set.exe
+random seed: 429006728
+generated error fail pass / total     time test name
+[✗]    1    0    1    0 / 1000     1.9s Lin parallel tests
+
+--- Failure --------------------------------------------------------------------
+
+Test Lin parallel tests failed (41 shrink steps):
+
+                         |
+                         |
+              .---------------------.
+              |                     |
+           add 0 t               add 0 t
+                                cardinal t
+
+
++++ Messages ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+Messages for test Lin parallel tests:
+
+  Results incompatible with sequential execution
+
+                              |
+                              |
+           .------------------------------------.
+           |                                    |
+      add 0 t : ()                         add 0 t : ()
+                                          cardinal t : 2
+
+================================================================================
+failure (1 tests failed, 0 tests errored, ran 1 tests)
+]}
+
+In the case of a failing test, [qcheck-lin] gives the user the counter example
+it has found, after some shrinking step, as it is customary in property-based
+testing {i à la} QuickCheck. The counter example is a program with a sequential
+prefix and two parallel suffixes. The counter example is given twice: the first
+time with just the function calls; and the second time, each call is paired
+with the result of the computation.
+
+Here, the counter example is composed of an empty sequential prefix and two
+spawned processes. The first one add 0 to the set, while the second one also
+add 0 to the set and then ask for the cardinality of the said set.
+
+In terms of sequential interleaving, there are only three possibilities. If we
+prefix the function call by [A] and [B] depending on the process in which they
```

Comment:
Would `[Left]` and `[Right]` be a better match to the printed ASCII diagram with two columns?